aten/src/ATen/native/Distance.cpp

```diff
@@ -75,4 +75,82 @@ Tensor cosine_similarity(const Tensor& x1, const Tensor& x2, int64_t dim, double
   return w12.div_((w1 * w2).clamp_min_(eps));
 }
 
+Tensor to_squareform(const Tensor& self) 
+{
+  AT_CHECK(self.dim() == 1,
+    "expected vector-form distance as input, dim as 1, but got ", self.dim());
```

Comment:
this error message is a little awkward.  How about something like "Expected 1-dimensional (vector-form) distance, but got tensor of dimensionality "...